### ISSUE TITLE
Should client_credentials have ability to include refresh token ?

### ISSUE THREAD

**Issue by JonathanHuot** [oauthlib/oauthlib#552]
Hi, I was looking on the [OAuth2 RFC about the client_credentials/Access Token Response](https://tools.ietf.org/html/rfc6749#section-4.4.3), and it says `A  refresh token SHOULD NOT be included.` So its inclusion depends of the implementor's  decision.

However, in `oauthlib` we don't give the choice to the implementor. It force the `refresh_token` to NOT be generated. See https://github.com/oauthlib/oauthlib/blob/master/oauthlib/oauth2/rfc6749/grant_types/client_credentials.py#L77 

In my opinion, `oauthlib` should set the _default_ as **to not include a refresh token**, but can still give the possibility to the implementor to generate it, if wanted.

Any comments?

**Comment by IvanAnishchuk** [comment]
Yeah, totally. `SHOULD` means it's not forbidden to include it and some implementations may choose to include it, for example if it's beneficial to track token refreshing for client credentials clients. And compliant clients should ignore any response parameters they don't understand so there should be no problem there too.

It'd also change `BackendApplicationClient` so it can emulate "refreshing" by simply getting a new token, it only requires a couple lines and would make api more uniform.
